File: openapi-generator/dart-jaguar/lib/model/vehicle_harsh_event_response.dart
import 'package:jaguar_serializer/jaguar_serializer.dart';

import 'package:openapi/model/vehicle_harsh_event_response_location.dart';
part 'vehicle_harsh_event_response.jser.dart';

class VehicleHarshEventResponse {
   /* URL for downloading the forward facing video */
  @Alias('downloadForwardVideoUrl')
  final String downloadForwardVideoUrl;
   /* URL for downloading the inward facing video */
  @Alias('downloadInwardVideoUrl')
  final String downloadInwardVideoUrl;
   /* URL for downloading the tracked inward facing video */
  @Alias('downloadTrackedInwardVideoUrl')
  final String downloadTrackedInwardVideoUrl;
   /* Type of the harsh event. One of: [Crash, Harsh Acceleration, Harsh Braking, Harsh Turn, ROP Engine, ROP Brake, YC Engine, YC Brake, Harsh Event] */
  @Alias('harshEventType')
  final String harshEventType;
   /* URL of the associated incident report page */
  @Alias('incidentReportUrl')
  final String incidentReportUrl;
   /* Whether the driver was deemed distracted during this harsh event */
  @Alias('isDistracted')
  final bool isDistracted;
  
  @Alias('location')
  final VehicleHarshEventResponseLocation location;
  

  VehicleHarshEventResponse(
      

{
     this.downloadForwardVideoUrl = null,  
     this.downloadInwardVideoUrl = null,  
     this.downloadTrackedInwardVideoUrl = null,  
    
     this.harshEventType = null,  
     this.incidentReportUrl = null,   this.isDistracted = null,  
     this.location = null 
    
    }
  );

  @override
  String toString() {
    return 'VehicleHarshEventResponse[downloadForwardVideoUrl=$downloadForwardVideoUrl, downloadInwardVideoUrl=$downloadInwardVideoUrl, downloadTrackedInwardVideoUrl=$downloadTrackedInwardVideoUrl, harshEventType=$harshEventType, incidentReportUrl=$incidentReportUrl, isDistracted=$isDistracted, location=$location, ]';
  }
}

@GenSerializer(nullableFields: true)
class VehicleHarshEventResponseSerializer extends Serializer<VehicleHarshEventResponse> with _$VehicleHarshEventResponseSerializer {

}

File: openapi-generator/php-ze-ph/src/App/DTO/VehicleHarshEventResponse.php
<?php
declare(strict_types=1);

namespace App\DTO;

use Articus\DataTransfer\Annotation as DTA;

/**
 * Harsh event details for a vehicle
 */
class VehicleHarshEventResponse
{
    /**
     * URL for downloading the forward facing video
     * @DTA\Data(field="downloadForwardVideoUrl", nullable=true)
     * @DTA\Validator(name="Type", options={"type":"string"})
     * @var string
     */
    public $download_forward_video_url;
    /**
     * URL for downloading the inward facing video
     * @DTA\Data(field="downloadInwardVideoUrl", nullable=true)
     * @DTA\Validator(name="Type", options={"type":"string"})
     * @var string
     */
    public $download_inward_video_url;
    /**
     * URL for downloading the tracked inward facing video
     * @DTA\Data(field="downloadTrackedInwardVideoUrl", nullable=true)
     * @DTA\Validator(name="Type", options={"type":"string"})
     * @var string
     */
    public $download_tracked_inward_video_url;
    /**
     * Type of the harsh event. One of: [Crash, Harsh Acceleration, Harsh Braking, Harsh Turn, ROP Engine, ROP Brake, YC Engine, YC Brake, Harsh Event]
     * @DTA\Data(field="harshEventType")
     * @DTA\Validator(name="Type", options={"type":"string"})
     * @var string
     */
    public $harsh_event_type;
    /**
     * URL of the associated incident report page
     * @DTA\Data(field="incidentReportUrl")
     * @DTA\Validator(name="Type", options={"type":"string"})
     * @var string
     */
    public $incident_report_url;
    /**
     * Whether the driver was deemed distracted during this harsh event
     * @DTA\Data(field="isDistracted", nullable=true)
     * @DTA\Validator(name="Type", options={"type":"bool"})
     * @var bool
     */
    public $is_distracted;
    /**
     * @DTA\Data(field="location", nullable=true)
     * @DTA\Strategy(name="Object", options={"type":\App\DTO\VehicleHarshEventResponseLocation::class})
     * @DTA\Validator(name="Dictionary", options={"type":\App\DTO\VehicleHarshEventResponseLocation::class})
     * @var \App\DTO\VehicleHarshEventResponseLocation
     */
    public $location;
}
